### DISCUSSION TITLE
KKP - failed to add finalizer

### DISCUSSION THREAD

**Discussion by clickersmudge** [kubermatic/kubermatic#11244] [Q&A] [answered]
Hi,
After creating a project and a new cluster, such a log is displayed in the KKP panel (v2.21.2), Hetzner.

![Zrzut ekranu z 2022-10-27 21-24-21](https://user-images.githubusercontent.com/113194014/198380386-e72c4232-002b-4145-9870-f28a463a0c07.png)

```
failed to add finalizer: failed to add finalizers [kubermatic.k8c.io/cleanup-credentials-secrets] to Cluster /izq9wfzv97: Operation cannot be fulfilled on clusters.kubermatic.k8c.io "izq9wfzv97": the object has been modified; please apply your changes to the latest version and try again```

I don't understand the reason.

**Comment by embik** [comment, marked as answer]
Hi @clickersmudge, this kind of event can be logged because during cluster creation several controller loops work on the fresh cluster at the same time and the cluster object gets updated frequently during its early stage. Since KKP follows Kubernetes design principles, a failed reconcile is usually retried, and this error message should not be an issue unless the count continuously goes up.

This is unfortunately something that occasionally happens with loosely coupled controller loops but has no impact on your cluster's health. If you want to verify, you can run `kubectl get cluster <Cluster ID> -o yaml` on your Seed and verify that the mentioned finalizer is in the list of finalizers.

**Reply by clickersmudge** [reply]
Thank you @embik.
